database/src/main/scala/cromwell/database/slick/MetadataComponent.scala

```diff
@@ -0,0 +1,62 @@
+package cromwell.database.slick
+
+
+import java.sql.Timestamp
+
+import cromwell.database.obj.Metadatum
+
+trait MetadataComponent {
+  this: DriverComponent with WorkflowExecutionComponent =>
+
+  import driver.api._
+
+  class Metadata(tag: Tag) extends Table[Metadatum](tag, "METADATA_JOURNAL") {
+    def metadataId = column[Int]("METADATA_JOURNAL_ID", O.PrimaryKey, O.AutoInc)
+    def workflowExecutionUuid = column[String]("WORKFLOW_EXECUTION_UUID")
+    def key = column[String]("METADATA_KEY")
+    def callFqn = column[Option[String]]("METADATA_CALL_FQN")
+    def index = column[Option[Int]]("METADATA_CALL_INDEX")
+    def attempt = column[Option[Int]]("METADATA_CALL_ATTEMPT")
+    def value = column[Option[String]]("METADATA_VALUE")
+    def timestamp = column[Timestamp]("METADATA_TIMESTAMP")
+
+    override def * = (workflowExecutionUuid, key, callFqn, index, attempt, value, timestamp, metadataId.?) <>
+      (Metadatum.tupled, Metadatum.unapply)
+  }
+
+  protected val metadata = TableQuery[Metadata]
+
+  val metadataByWorkflowUuid = Compiled(
+    (workflowExecutionUuid: Rep[String]) => for {
+      metadatum <- metadata
+      if metadatum.workflowExecutionUuid === workflowExecutionUuid
+    } yield metadatum)
+
+  val metadataByWorkflowUuidAndKey = Compiled(
+    (workflowExecutionUuid: Rep[String], key: Rep[String]) => for {
+      metadatum <- metadata
+      if metadatum.workflowExecutionUuid === workflowExecutionUuid
+      if metadatum.key === key
+    } yield metadatum)
```

Comment:
Now that I re-read this I'm not sure it's doing the right thing.  Should this be returning only workflow-scoped KVs (i.e. KVs with null callFqn, index, and scope) or every KV in this workflow having this key?  It'd doing the latter, but I'm not sure the former wasn't the intent of the API.